src/Ryujinx.Ui.Common/App/ApplicationLibrary.cs

```diff
@@ -508,22 +508,43 @@ private void GetControlFsAndTitleId(PartitionFileSystem pfs, out IFileSystem con
             titleId   = controlNca?.Header.TitleId.ToString("x16");
         }
 
-        public ApplicationMetadata LoadAndSaveMetaData(string titleId, Action<ApplicationMetadata> modifyFunction = null)
+        public ApplicationMetadata LoadAndSaveMetaData(UserId userId, string titleId, Action<ApplicationMetadata> modifyFunction = null)
         {
-            string metadataFolder = Path.Combine(AppDataManager.GamesDirPath, titleId, "gui");
+            string guiFolder      = Path.Combine(AppDataManager.GamesDirPath, titleId, "gui");
+            string metadataFolder = Path.Combine(guiFolder, userId.ToString());
             string metadataFile   = Path.Combine(metadataFolder, "metadata.json");
 
+            if (!Directory.Exists(metadataFolder))
+            {
+                Directory.CreateDirectory(metadataFolder);
+            }
```

Comment:
As per [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-7.0):
`Any and all directories specified in path are created, unless they already exist or unless some part of path is invalid. If the directory already exists, this method does not create a new directory, but it returns a DirectoryInfo object for the existing directory.`

```suggestion
            Directory.CreateDirectory(metadataFolder);
```

Reply:
Thank you ! Not a .NET expert for sure 😅  I will take this part in mind.